File: Server/services/Product.service.js
import ProductRepository from '../repositories/Product.repository.js';
const ProductServices = {
  async changeStatus(id, status) {
    return await ProductRepository.changeStatus(id, status);
  },
  async changeIsDisable(id, is_disable) {
    return await ProductRepository.changeIsDisable(id, is_disable);
  },
  async create(
    name,
    description,
    size,
    price_S,
    price_M,
    price_L,
    note,
    image,
    status,
    category_id,
    update_date,
    release_date,
    sales
  ) {
    return await ProductRepository.create(
      name,
      description,
      size,
      price_S,
      price_M,
      price_L,
      note,
      image,
      status,
      category_id,
      update_date,
      release_date,
      sales
    );
  },
  async update(
    id,
    name,
    description,
    size,
    price_S,
    price_M,
    price_L,
    image,
    status,
    category_id,
    update_date,
    release_date,
    sales
  ) {
    return await ProductRepository.update(
      id,
      name,
      description,
      size,
      price_S,
      price_M,
      price_L,
      image,
      status,
      category_id,
      update_date,
      release_date,
      sales
    );
  },
  async delete(id) {
    return await ProductRepository.delete(id);
  },
  async findOne(id) {
    return await ProductRepository.findOneByID(id);
  },
  async findAll() {
    return await ProductRepository.findAll();
  },
  async findAllFavProduct(userId) {
    return await ProductRepository.findAllFavProduct(userId)
  },
  async addNewFavProduct(userId, productId) {
    return await ProductRepository.addNewFavProduct(userId, productId)
  },
  async removeFavProduct(userId, productId) {
    return await ProductRepository.removeFavProduct(userId, productId)
  },
  async isExistedFavProduct(userId, productId) {
    return await ProductRepository.isExistedFavProduct(userId, productId)
  },
  async getRating(productId) {
    return await ProductRepository.getRating(productId)
  }
  // Cần method gì thì tự implements !!
};
export default ProductServices;

File: App User/app/src/main/java/com/example/myapplication/Admin/services/ProductService.kt
package com.example.myapplication.Admin.services

import com.example.myapplication.Admin.modals.Product
import okhttp3.MultipartBody
import okhttp3.RequestBody
import retrofit2.http.*


interface ProductService {
    @GET("product/{id}")
    suspend fun getProduct(@Path("id") productId: Int): Product

    @GET("product/")
    suspend fun getAllProduct(): List<Product>

    @Multipart
    @POST("product/create")
    suspend fun createProduct(
        @Part("name") name: RequestBody,
        @Part("description") description: RequestBody,
        @Part("size") size: RequestBody,
        @Part("price_S") price_S: RequestBody,
        @Part("price_M") price_M: RequestBody,
        @Part("price_L") price_L: RequestBody,
        @Part("status") status: RequestBody,
        @Part("category_id") category_id: RequestBody,
        @Part("update_date") update_date: RequestBody,
        @Part("release_date") release_date: RequestBody,
        @Part("sales") sales: RequestBody,
        @Part image: MultipartBody.Part
    ): Product

    @Multipart
    @PUT("product/update/{id}")
    suspend fun updateProduct(
        @Path("id") productId: Int,
        @Part("name") name: RequestBody,
        @Part("description") description: RequestBody,
        @Part("size") size: RequestBody,
        @Part("price_S") price_S: RequestBody,
        @Part("price_M") price_M: RequestBody,
        @Part("price_L") price_L: RequestBody,
        @Part("category_id") category_id: RequestBody,
        @Part("update_date") update_date: RequestBody,
        @Part image: MultipartBody.Part
    ): Product

    @PUT("product/updateWithoutImage/{id}")
    suspend fun updateWithoutImage(
        @Path("id") productId: Int,
        @Body product: Product
    ): Product

    @DELETE("product/delete/{id}")
    suspend fun deleteProduct(@Path("id") productId: Int): Boolean

    @PUT("product/disable/{id}")
    suspend fun disableProduct(@Path("id") productId: Int): Boolean

    @PUT("product/enable/{id}")
    suspend fun enableProduct(@Path("id") productId: Int): Boolean

    @PUT("product/available/{id}")
    suspend fun availableProduct(@Path("id") productId: Int): Boolean

    @PUT("product/unavailable/{id}")
    suspend fun unavailableProduct(@Path("id") productId: Int): Boolean
}